src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,194 @@
+                        // Completed task
+                        Task task = new Deadline(arr[2], arr[3]);
+                        task.markAsDone();
+                        taskList.add(task);
+                    } else {
+                        // Not recognised format
+                        Ui.printFormatError(i);
+                    }
+                } catch (Exception ex) {
+                    Ui.printFormatError(i);
+                }
+                break;
+            case "E":
+                // It is an duke.Event task
+                try {
+                    if (Integer.valueOf(arr[1]).equals(0)) {
+                        // Incomplete task
+                        Task task = new Event(arr[2], arr[3]);
+                        taskList.add(task);
+                    } else if (Integer.valueOf(arr[1]).equals(1)) {
+                        // Completed task
+                        Task task = new Event(arr[2], arr[3]);
+                        task.markAsDone();
+                        taskList.add(task);
+                    } else {
+                        // Not recognised format
+                        Ui.printFormatError(i);
+                    }
+                } catch (Exception ex) {
+                    Ui.printFormatError(i);
+                }
+                break;
+            default:
+                // No such type
+                Ui.printFormatError(i);
+                break;
+            }
+        }
+        return taskList;
+    }
+
+    /**
+     * Save and write the file into todolist.txt.
+     * @param f File The file.
+     * @param ls duke.TaskList the list of the tasks to be written.
+     * @throws IOException On input error.
+     */
+    protected static void saveFile(File f, TaskList ls) throws IOException {
+        ArrayList<String> arrayList = new ArrayList<>();
+        String link = " >> ";
+        FileWriter fw = new FileWriter(f.getAbsolutePath());
+
+        if (ls.size() == 0) {
+            fw.write("");
+        } else {
+            for (int i = 0; i < ls.size(); i++) {
+                Task task = ls.get(i);
+                if (task instanceof Todo) {
+                    String str;
+                    if (!task.isDone()) {
+                        str = "T" + link + "0" + link + task.getDescription();
+                    } else {
+                        str = "T" + link + "1" + link + task.getDescription();
+                    }
+                    arrayList.add(str);
+
+                } else if (task instanceof Event) {
+                    String str;
+                    if (!task.isDone()) {
+                        str = "E" + link + "0" + link + task.getDescription() + link
+                                + ((Event) task).getAt().toString();
+                    } else {
+                        str = "E" + link + "1" + link + task.getDescription() + link
+                                + ((Event) task).getAt().toString();
+                    }
```

Comment:
One way to reduce this duplicate code would be to overload the toString method in your TaskList class.